src/pool/srv_pool.c

```diff
@@ -3628,6 +3765,355 @@ ds_pool_prop_set_handler(crt_rpc_t *rpc)
 	crt_reply_send(rpc);
 }
 
+static int pool_upgrade_props(struct rdb_tx *tx, struct pool_svc *svc,
+			      uuid_t pool_uuid, crt_rpc_t *rpc)
+{
+	d_iov_t			value;
+	uint64_t		val;
+	uint32_t		val32;
+	int			rc;
+	bool			need_commit = false;
+	uuid_t		       *hdl_uuids = NULL;
+	size_t			hdl_uuids_size;
+	int			n_hdl_uuids = 0;
+
+	if (rpc) {
+		rc = find_hdls_to_evict(tx, svc, &hdl_uuids, &hdl_uuids_size,
+					&n_hdl_uuids, NULL);
+		if (rc)
+			return rc;
+		D_DEBUG(DB_MGMT, "number of handles found was: %d\n", n_hdl_uuids);
+	}
+
+	if (n_hdl_uuids > 0) {
+		rc = pool_disconnect_hdls(tx, svc, hdl_uuids, n_hdl_uuids,
+					  rpc->cr_ctx);
```

Comment:
Do we also need to set ds_pool_prop_connectable to 1 before calling pool_disconnect_hdls and set it back to 0 when the upgrade terminates (successfully or unsuccessfully), so that no new connection will be established during the upgrade?

Reply:
Yes, i thought i handled  this logic in the above function, i should do this.